nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp

```diff
@@ -275,6 +275,22 @@ class BtActionNode : public BT::ActionNodeBase
       }
```

Comment:
> When you run into issues, are you seeing that print?

No, we don't see it. 
Actually, the problem we would like to fix is 
**not** that the result of the **cancel** request is not available or has failed (which would be reported by *that print*, indeed), 
**but** that the result of the **goal** request sent to the sub-action is not always available before halting the whole BT. 
I may have not been so clear in the PR description, I will edit it right away in order to replace every "result" reference by either "goal result" or "cancel result".

> If the problem as you describe it is correct, you should be such that would return a non-success return code

I think there is a misunderstanding, because of the result we are discussing here: you mean the **cancel** result (right?), while I meant the **goal** result is *the problem*. (By *the problem*, I mean *when the **goal** result is not available*)
So, if the *non-success return code* you are talking about here is the return code of the result of the **cancel** request: well no, *the problem* happens even if the result of the cancel request is `SUCCESS`. 
Actually, in 100% of the occurrences of *the problem*, the result of the **cancel** request was `SUCCESS`.

>  I think given the code we're adding here, then that error should be reduced to a warning or a debug since we're going to be retrying below.

No, sorry I don't agree, we are not *retrying* anything in this `halt()` function. 
For me to be clearer, let's split the `halt()` function into 3 big steps:
- The first step (already implemented) consists in [sending the **cancel** request (L268)](https://github.com/ros-planning/navigation2/blob/29c1af087afffb80ddc4ec172015fee2c8a37166/nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp#L268) and [checking for the result of this **cancel** request (L269)](https://github.com/ros-planning/navigation2/blob/29c1af087afffb80ddc4ec172015fee2c8a37166/nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp#L269-L275).
- The second step (the subject of this PR) would consist in [checking for the result of the **goal** request (L279)](https://github.com/wyca-robotics/navigation2/blob/04261fb7633c69baf709bae739310e37a3322812/nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp#L279-L287)
- Finally the third step (already implemented) simply consists in [applying the `IDLE` state to the BT node](https://github.com/ros-planning/navigation2/blob/29c1af087afffb80ddc4ec172015fee2c8a37166/nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp#L278).

So the outcomes of the steps 1 and 2 are not related: the first one can end on a success and not the second one, or the opposite, or both can fail, or both can success.
The step 2 (this PR) is here to prevent from reaching step 3 without having the result of the **goal** request available.

Reply:
Got it, so it was problem (B) of the 2 I mention in the comments below? If so, then we can resolve this comment, no action required. This applies only to problem (A)